app_dart/lib/src/service/datastore.dart

```diff
@@ -147,4 +151,25 @@ class DatastoreService {
       return previousStatusUpdates.single;
     }
   }
+
+  /// Generate new authorization token for [agent]
+  ///
+  /// The hashed code of token will be returned as a list
+  List<int> refreshAgentAuthToken(Agent agent){
```

Comment:
This doesn't do any datastore access, so it's a little out of place in `DatastoreService`.  Maybe create a new `AgentService`?

Reply:
Put it here only because this function was defined in db.go from GO side. Yeah, it makes sense to have a separate service for agent. Just created new AgentService as suggested.